django_opentracing/tracer.py

```diff
@@ -59,43 +71,57 @@ def wrapper(request):
     def _apply_tracing(self, request, view_func, attributes):
         '''
         Helper function to avoid rewriting for middleware and decorator.
-        Returns a new span from the request with logged attributes and 
+        Returns a new span from the request with logged attributes and
         correct operation name from the view_func.
         '''
         # strip headers for trace info
         headers = {}
-        for k,v in request.META.iteritems():
-            k = k.lower().replace('_','-')
+        for k, v in request.META.iteritems():
+            k = k.lower().replace('_', '-')
             if k.startswith('http-'):
                 k = k[5:]
-            headers[k] = v              
+            headers[k] = v
 
         # start new span from trace info
         span = None
         operation_name = view_func.__name__
+        tags = SERVER_SPAN_TAGS.copy()
+        tags[ot_tags.HTTP_METHOD] = request.method
+        tags[ot_tags.HTTP_URL] = request.build_absolute_uri()
         try:
             span_ctx = self._tracer.extract(opentracing.Format.HTTP_HEADERS, headers)
-            span = self._tracer.start_span(operation_name=operation_name, child_of=span_ctx)
-        except (opentracing.InvalidCarrierException, opentracing.SpanContextCorruptedException) as e:
-            span = self._tracer.start_span(operation_name=operation_name)
+            span = self._tracer.start_span(operation_name=operation_name,
+                                           child_of=span_ctx, tags=tags)
+        except (opentracing.InvalidCarrierException,
+                opentracing.SpanContextCorruptedException):
+            span = self._tracer.start_span(operation_name=operation_name,
+                                           tags=tags)
         if span is None:
-            span = self._tracer.start_span(operation_name=operation_name)
-
-        # add span to current spans 
-        self._current_spans[request] = span
+            span = self._tracer.start_span(operation_name=operation_name,
+                                           tags=tags)
 
         # log any traced attributes
         for attr in attributes:
             if hasattr(request, attr):
                 payload = str(getattr(request, attr))
                 if payload:
                     span.set_tag(attr, payload)
-        
-        return span  
+
+        start_hook = self.hooks.get('start')
+        if start_hook and callable(start_hook) and start_hook(span):
+            return span
+
+        # add span to current spans
+        self._current_spans[request] = span
```

Comment:
Yes. When I was fixing the merge conflicts I kind of looked at that specific line. It does seem like it should be above the `if start_hook and callable(start_hook) and start_hook(span)` conditional, but I was just updating things as it was. Also, this could potentially have problems which I've addressed with #20. However, I'm going to adopt the same tag names for #20 as they do in [opentracing-python here](https://github.com/opentracing/opentracing-python/blob/5bf9527c60168bece53bf0fc437f43a2a55dd46c/opentracing/span.py#L200-L202) and will likely put most of the code inside trace.py and not inside the middleware. Should be able to get both of those done tonight. What do you think?